types/yargs-parser/index.d.ts

```diff
@@ -2,12 +2,12 @@
 // Project: https://github.com/yargs/yargs-parser#readme
 // Definitions by: Miles Johnson <https://github.com/milesj>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.2
+// TypeScript Version: 2.8
 
 import { Arguments as YargsArguments } from 'yargs';
 
 declare namespace yargsParser {
-    type Arguments = YargsArguments;
```

Comment:
It would be nice if the generic was passed in via `Parser`.

Reply:
The typings for yargs-parser could indeed be improved using the same techniques as in this PR, but there will be some differences since the parser is more configurable at this lower level. At the same time, you're passing all options in one go here (on line 52-53), instead of chaining multiple calls, so that part is less complex.

For now, I just made the minimal required change here, since YargsArguments gained a type parameter, and you can't create a type alias without specifying the type parameters (even if the type parameters have defaults).

I'm not sure how common it is to use yargs-parser directly and not through yargs. But I can do a follow-up PR to update yargs-parser as well, if there is interest in that.